### PR TITLE
Kubernetes default ingress

### LINKED ISSUE #81: Add an option in Kubernetes Cluster to create default IngressNginx instance
**Is your feature request related to a problem? Please describe.**

Users want to publish applications with default Ingress Controllers in Kubernetes Cluster.

**Describe the solution you'd like**

Add an option in Kubernetes Cluster to create default IngressNginx instance in the ClusterAddons.

### PR DESCRIPTION
**Pull request by pasqualet**
**What type of PR is this?**

/kind bug
/kind feature

**What this PR does / why we need it**

Add an option in Kubernetes Cluster to create default IngressNginx instance.

**Which issue(s) this PR fixes**

Fixes #81 

**Special notes for your reviewer**

**Does this PR introduce a user-facing change?**

```release-note
Add default IngressController in Kubernetes Cluster
Fixed calico race condition
Always redirect http to https in Kubernetes Cluster IngressController
```

### CODE REVIEW

**Review by bernardomp — CHANGES_REQUESTED** [review]
La clases de Ingress funciona ok 
image

Por otro lado,  parece que el parámetro public no funciona correctamente. Al menos, cuando se cambia a true o false,  no desencana ningun cambio a nivel de pulumi 

const cluster1 = new ct.Cluster("my-eks-ber",{networking: {

    admin: {
        domain: "apps.astrokube.es",
        public: false,
    },
    default: {
        domain: "default.astrokube.es",
        public: false,
    }

}});

**Comment by pasqualet** [comment]
> La clases de Ingress funciona ok image
> 
> Por otro lado, parece que el parámetro public no funciona correctamente. Al menos, cuando se cambia a true o false, no desencadena ningún cambio a nivel de pulumi
> 
> ```
> const cluster1 = new ct.Cluster("my-eks-ber",{networking: {
>     admin: {
>         domain: "apps.astrokube.es",
>         public: false,
>     },
>     default: {
>         domain: "default.astrokube.es",
>         public: false,
>     }
> }});
> ```

You are right. I've fixed the issue with another commit. Anyway, even if we change the pulumi configuration, the LoadBalancer in AWS can't be changed. So it's a value only valid when the cluster is created. In the future 

> La clases de Ingress funciona ok image
> 
> Por otro lado, parece que el parámetro public no funciona correctamente. Al menos, cuando se cambia a true o false, no desencadena ningún cambio a nivel de pulumi
> 
> ```
> const cluster1 = new ct.Cluster("my-eks-ber",{networking: {
>     admin: {
>         domain: "apps.astrokube.es",
>         public: false,
>     },
>     default: {
>         domain: "default.astrokube.es",
>         public: false,
>     }
> }});
> ```

You are right. The public property wasn't propagated from the Cluster to the ClusterAddons component. Fixed in the new commit.

Anyway, even if Pulumi apply the changes, the LoadBalancer will not change to private/public because of AWS. So this can be configured only when the cluster is created, at least for now. In the future we have to recreate the IngressController component when this field changes.

File: provider/cmd/pulumi-resource-cloud-toolkit-aws/kubernetes/clusterAddons.ts [hunk 1]
```diff
@@ -71,6 +77,11 @@ export class ClusterAddons extends pulumi.ComponentResource {
    */
   public readonly awsLoadBalancerController: AwsLoadBalancerController;
 
+  public adminZoneArn?: pulumi.Input<string>;
```
[the remark this answers is not in the document]
Comment (pasqualet) [reply provider/cmd/pulumi-resource-cloud-toolkit-aws/kubernetes/clusterAddons.ts]: Fixed!

**Review by bernardomp — COMMENTED** [review]
OK, now pulumi detects changes and ingress class works.

image

**Review by bernardomp — APPROVED** [review]

**Pull request merged**